com.unity.render-pipelines.universal/Documentation~/upgrade-guide-2022-1.md

```diff
@@ -0,0 +1,218 @@
+# Upgrading to version 2022.1 of the Universal Render Pipeline
+
+This page describes how to upgrade from an older version of the Universal Render Pipeline (URP) to version 2022.1.
+
+## Upgrading from URP 2021.2
+
+* An error will be issued when instances of `ScriptableRendererFeature` attempt to access render targets before they are allocated by the renderer.
+
+   The `ScriptableRendererFeature` has a new virtual function called `SetupRenderPasses` which is called when render targets are allocated and ready to be used.
+
+  If your code uses `renderer.cameraColorTarget` or `renderer.cameraDepthTarget` inside of the `AddRenderPasses` override, then that use needs to be moved to `SetupRenderPasses`.
+  Note that calls to `renderer.EnqueuePass` should still happen in `AddRenderPasses`.
+
+  For example the following use:
+  ```c#
+  public override void AddRenderPasses(ScriptableRenderer renderer, ref RenderingData renderingData)
+  {
+      m_CustomPass.Setup(renderer.cameraColorTarget);  // use of target before allocation
+      renderer.EnqueuePass(m_ScriptablePass); // letting the renderer know which passes will be used before allocation
+  }
+  ```
+
+  should become:
+
+  ```c#
+  public override void AddRenderPasses(ScriptableRenderer renderer, ref RenderingData renderingData)
+  {
+      renderer.EnqueuePass(m_ScriptablePass); // letting the renderer know which passes will be used before allocation
+  }
+
+  public override void SetupRenderPasses(ScriptableRenderer renderer, in RenderingData renderingData)
+  {
+      m_CustomPass.Setup(renderer.cameraColorTarget);  // use of target after allocation
+  }
+  ```
+
+* The Universal Renderer is now using [`RTHandles`](https://docs.unity3d.com/Packages/com.unity.render-pipelines.core@13.1/manual/rthandle-system.html) for its internal targets and in its internal passes.
+
+  All uses of `RenderTargetHandle` have been set as Obsolete and the class will be removed in the future.
+
+  The public interfaces `renderer.cameraColorTarget` and `renderer.cameraDepthTarget` have also been marked as obsolete and their uses should be replaced with  `renderer.cameraColorTargetHandle` and `renderer.cameraDepthTargetHandle` respectively.
+
+  `RTHandle` targets do not use `GetTemporaryRT` and are longer-lived than the `RenderTargetIdentifiers` from there. They also cannot be allocated with a `GraphicsFormat` and `DepthBufferBits` set to anything but 0. Depth targets must be separate from Color Targets.
+
+  The following helper functions have been added in order to create and use `RTHandle` targets in the same way as with `GetTemporaryRT`:
+     * `RenderingUtils.ReAllocateIfNeeded`
+     * `ShadowUtils.ShadowRTReAllocateIfNeeded`
```

Comment:
These don't really feel like helper functions, but more like something you _need_ to use. Can we move them into a more official sounding class?